package_scanner.nim

```diff
@@ -1,135 +1,367 @@
+      if url notin urls:
+        urls.incl url
+      else:
+        fatal("Package by that URL already exists: " & $url)
+
+  test "Check Alias":
+    doAssert names.len > 0, $names
+    doAssert aliasList.len > 0, $aliasList
+    for pdata in aliasList:
+      var alias = pdata["alias"].str
+      check alias.len == alias.strip.len
+      check alias.strip.toLowerAscii in names  # Alias must relate to a name
+
+  test "Check Profanity Filter by Levenshtein Distance":
+    when checkProfanity:
+      for pdata in pckgsList:
+        # Name
+        for badword in profanities:
+          check editDistanceAscii(pdata["name"].str.normalize, badword) >= profanityLevensteinMax
+        # Tags
+        for tag in pdata["tags"]:
+          for badword in profanities:
+            check editDistanceAscii(tag.str.normalize, badword) >= profanityLevensteinMax
+        # Description
+        for description in pdata["description"].str.normalize.splitWhitespace:
+          for badword in profanities:
+            check editDistanceAscii(description, badword) >= profanityLevensteinMax
+
+  test "Check URLs On-Line by Git":
+    when checkByGit:
+      for pdata in pckgsList:
+        removeDir(gitTempDir)
+        check execCmd(gitCmd.format(pdata["url"].str)) == 0
+
+  test "Check URLs On-Line by HttpClient":
+    when defined(ssl):
+      var existent, nonexistent, nimble_existent, nimble_nonexistent: seq[string]
+      for pdata in pckgsList:
+        var
+          skip: bool
+          url = pdata["url"].str.strip.toLowerAscii
+          name = pdata["name"].str.normalize
 
-  echo ""
-  echo "Problematic packages count: ", result
+        # Some hostings randomly timeout or fail sometimes, skip them.
+        for skipurl in hostsSkip:
+          if url.startsWith(skipurl):
+            skip = true
+        if skip: continue
 
+        echo url  # Do Not remove, Travis quits with "N minutes without output, job cancelled".
+
+        # Check that the Git Repo actually exists.
+        var this_repo = existsGitRepo(url=url) # Fetch test.
+        if this_repo.len > 0:
+          existent.add this_repo
+        else:
+          nonexistent.add url
+
+        # Check for Nimble Files on Existent Repos.
+        if this_repo.len > 0 and checkNimbleFile:
+          var this_nimble = existsNimbleFile(url=this_repo, name=name)
+          if this_nimble.len > 0:
+            nimble_existent.add this_nimble
+          else:
+            nimble_nonexistent.add url
+
+      # Warn or Assert the possible errors at the end.
+      if nonexistent.len > 0 and allowBrokenUrl:
+        warn "Missing repos list:\n" & nonexistent.join("\n  ")
+        warn "Missing repos count: " & $nonexistent.len & " of " & $pckgsList.len
+      else:
+        doAssert nonexistent.len == 0, "Missing repos: Broken Packages."
+      if nimble_nonexistent.len > 0 and allowMissingNimble:
+        warn "Missing Nimble files:\n" & nimble_nonexistent.join("\n  ")
```

Comment:
Nitpick: please stick to camelCase throughout.

Reply:
u right